machine/machine.go

```diff
@@ -173,51 +173,54 @@ func New(kvmPath string, nCpus int, tapIfName string, diskPath string, memSize i
 		return m, err
 	}
 
-	e, err := ebda.New(nCpus)
-	if err != nil {
-		return m, err
-	}
-
-	bytes, err := e.Bytes()
-	if err != nil {
-		return m, err
+	// Poison memory.
+	// 0 is valid instruction and if you start running in the middle of all those
+	// 0's it is impossible to diagnore.
+	for i := highMemBase; i < len(m.mem); i += len(Poison) {
+		copy(m.mem[i:], Poison)
 	}
 
-	copy(m.mem[bootparam.EBDAStart:], bytes)
+	return m, nil
+}
 
-	m.pci = pci.New(pci.NewBridge()) // 00:00.0 for PCI bridge
+func (m *Machine) AddPCI(dev pci.Device) {
+	m.pci = pci.New(dev)
+}
 
-	if len(tapIfName) > 0 {
-		t, err := tap.New(tapIfName)
-		if err != nil {
-			return nil, err
-		}
+func (m *Machine) AddTap(tapIfName string) error {
+	if m.pci == nil {
+		m.AddPCI(pci.NewBridge())
```

Comment:
This is confusing because it does not match the function name.
As for the pci bridge, basically one is enough, so I think we can just add it in `machine.New()`.